src/execution/join_hashtable.cpp

```diff
@@ -279,6 +280,13 @@ void JoinHashTable::Build(DataChunk &keys, DataChunk &payload) {
 
 	RowOperations::Scatter(source_chunk, source_data.data(), layout, addresses, *string_heap, *current_sel,
 	                       added_count);
+
+	if (!layout.AllConstant()) {
+		D_ASSERT(string_heap->keep_pinned);
+		for (size_t i = prev_rows_blocks; i < block_collection->blocks.size(); ++i) {
+			block_collection->blocks[i]->block->SetSwizzling("JoinHashTable::Build");
```

Comment:
If the heap blocks are being kept pinned, then I don't think the data blocks need to be tagged as unswizzled?

Reply:
You're right